ergotree-ir/src/types/sheader.rs

```diff
@@ -36,146 +66,97 @@ lazy_static! {
+        property("nBits", SType::SLong, N_BITS_METHOD_ID);
     static ref HEIGHT_PROPERTY_METHOD_DESC: SMethodDesc =
-        property("height", SType::SInt, HEIGHT_PROPERTY_METHOD_ID);
-}
-
-pub const EXTENSION_ROOT_PROPERTY_METHOD_ID: MethodId = MethodId(10);
-lazy_static! {
-    pub static ref EXTENSION_ROOT_PROPERTY: SMethod =
-        SMethod::new(Header, EXTENSION_ROOT_PROPERTY_METHOD_DESC.clone(),);
+        property("height", SType::SInt, HEIGHT_METHOD_ID);
     static ref EXTENSION_ROOT_PROPERTY_METHOD_DESC: SMethodDesc = property(
         "extensionRoot",
         SColl(SByte.into()),
-        EXTENSION_ROOT_PROPERTY_METHOD_ID
+        EXTENSION_ROOT_METHOD_ID
     );
-}
-
-pub const MINER_PK_PROPERTY_METHOD_ID: MethodId = MethodId(11);
-lazy_static! {
-    pub static ref MINER_PK_PROPERTY: SMethod =
-        SMethod::new(Header, MINER_PK_PROPERTY_METHOD_DESC.clone(),);
     static ref MINER_PK_PROPERTY_METHOD_DESC: SMethodDesc =
-        property("minerPk", SType::SGroupElement, MINER_PK_PROPERTY_METHOD_ID);
-}
-
-pub const POW_ONETIME_PK_PROPERTY_METHOD_ID: MethodId = MethodId(12);
-lazy_static! {
-    pub static ref POW_ONETIME_PK_PROPERTY: SMethod =
-        SMethod::new(Header, POW_ONETIME_PK_PROPERTY_METHOD_DESC.clone(),);
+        property("minerPk", SType::SGroupElement, MINER_PK_METHOD_ID);
     static ref POW_ONETIME_PK_PROPERTY_METHOD_DESC: SMethodDesc = property(
         "powOnetimePk",
         SType::SGroupElement,
-        POW_ONETIME_PK_PROPERTY_METHOD_ID
+        POW_ONETIME_PK_METHOD_ID
     );
-}
-
-pub const POW_NONCE_PROPERTY_METHOD_ID: MethodId = MethodId(13);
-lazy_static! {
-    pub static ref POW_NONCE_PROPERTY: SMethod =
-        SMethod::new(Header, POW_NONCE_PROPERTY_METHOD_DESC.clone(),);
     static ref POW_NONCE_PROPERTY_METHOD_DESC: SMethodDesc = property(
         "powNonce",
         SColl(SByte.into()),
-        POW_NONCE_PROPERTY_METHOD_ID
+        POW_NONCE_METHOD_ID
     );
-}
-
-pub const POW_DISTANCE_PROPERTY_METHOD_ID: MethodId = MethodId(14);
-lazy_static! {
-    pub static ref POW_DISTANCE_PROPERTY: SMethod =
-        SMethod::new(Header, POW_DISTANCE_PROPERTY_METHOD_DESC.clone(),);
     static ref POW_DISTANCE_PROPERTY_METHOD_DESC: SMethodDesc = property(
         "powDistance",
         SType::SBigInt,
-        POW_DISTANCE_PROPERTY_METHOD_ID
+        POW_DISTANCE_METHOD_ID
     );
+    static ref VOTES_PROPERTY_METHOD_DESC: SMethodDesc =
+        property("votes", SColl(SByte.into()), VOTES_METHOD_ID);
 }
 
-pub const VOTES_PROPERTY_METHOD_ID: MethodId = MethodId(15);
 lazy_static! {
-    pub static ref VOTES_PROPERTY: SMethod =
-        SMethod::new(Header, VOTES_PROPERTY_METHOD_DESC.clone(),);
-    static ref VOTES_PROPERTY_METHOD_DESC: SMethodDesc =
-        property("votes", SColl(SByte.into()), VOTES_PROPERTY_METHOD_ID);
+    /// Property used only for tests!!!
+    pub static ref UNKNOWN_PROPERTY: SMethod = SMethod::new(Header, property("unknown", SType::SByte, MethodId(100)),);
```

Comment:
I suggest moving it to eval::sheader::tests module.

Reply:
I should think about some workaround here, because `SMethod` has one private field and private for other crates constructors.
Or can I make `pub` what it's needed to move this to the stated module?

Reply:
Oh, please make `SMethod::new` constructor fully public then. I see no reason why I made it that way.